Source/Particles/Collision/BinaryCollision/DSMC/SplitAndScatterFunc.H

```diff
@@ -289,7 +277,23 @@ public:
 #endif
             }
 
-            // Next perform all product producing collisions
+            // Next perform all product-producing collisions
+            else if (mask[i] == int(ScatteringProcessType::CHARGE_EXCHANGE))
+            {
+                // create a copy of the first product species at the location of species 1
+                const auto product1_index = products_np_data[2] + with_product_p_offsets[i];
+                copy_species1[2](soa_products_data[2], soa_1, static_cast<int>(p_pair_indices_1[i]),
+                                static_cast<int>(product1_index), engine);
+                // Set the weight of the new particle to p_pair_reaction_weight[i]
+                soa_products_data[2].m_rdata[PIdx::w][product1_index] = p_pair_reaction_weight[i];
+
+                // create a copy of the other product species at the location of species 2
+                const auto product2_index = products_np_data[3] + with_product_p_offsets[i];
+                copy_species2[3](soa_products_data[3], soa_2, static_cast<int>(p_pair_indices_2[i]),
+                                static_cast<int>(product2_index), engine);
+                // Set the weight of the new particle to p_pair_reaction_weight[i]
+                soa_products_data[3].m_rdata[PIdx::w][product2_index] = p_pair_reaction_weight[i];
```

Comment:
We also talked last week about creating the first product species at the location of species 2 and the second product species at the location of species 1. This is meant to avoid "teleporting" charge between the two locations. I just wanted to make a note of that in case we wanted to address that in a follow-up PR.